src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -179,7 +179,8 @@ Copyright (c) .NET Foundation. All rights reserved.
       Interactive="$(NuGetInteractive)"
       RestoreForceEvaluate="$(RestoreForceEvaluate)"
       RestorePackagesConfig="$(RestorePackagesConfig)"
-      EmbedFilesInBinlog="$(RestoreEmbedFilesInBinlog)">
+      EmbedFilesInBinlog="$(RestoreEmbedFilesInBinlog)"
+      SDKAnalysisLevel="$(SDKAnalysisLevel)">
```

Comment:
Either this is not the right place to pass through the SDK analysis level, or we'll need a different design for Visual Studio and also NuGet.exe.

Within VS, we only get per-project information, there is no "whole solution" properties. In fact, Restore is quite unique, as I'm not aware of any other MSBuild task that runs at a solution level, everything else is per-project.  This also applies to `nuget.exe restore`, when we read PackageReference inputs.

Therefore, I'm fairly confident that we'll need to include this sdk analysis level in the PackageSpec, so that it's available to RestoreCommand, no matter which restore is being used ("normal" MSBuild restore, "static graph" restore, nuget.exe, VS).